.github/workflows/agent-sync-shared.yml

```
@@ -1,34 +1,39 @@
+        type: boolean
+
+env:
+  SYNC_ENVIRONMENT: ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.environment || 'Production' }}
+  SYNC_DRY_RUN: ${{ github.event.inputs.dry_run == true && 'true' || 'false' }}
+  SYNC_BACKUP_PREFIX: ${{ github.event_name == 'workflow_dispatch' && 'manual-agent-sync' || 'auto-agent-sync' }}
 
 jobs:
   backup:
     name: Backup Database
-    if: inputs.dry_run != 'true'
+    if: env.SYNC_DRY_RUN != 'true'
```

actionlint (1.7.7)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)